### DISCUSSION TITLE
Package Scheduling Functionality and existing PRs - Expected behaviour?

### DISCUSSION THREAD

**Discussion by danijam** [renovatebot/renovate#29886] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Kubernetes

### Please tell us more about your question or problem

I'm trying to implement a schedule for a particular package in a particular onboarded repo. However I was still getting updates pushed to an existing PR... I was expecting to not get updates

In reviewing the DEBUG log output. I can see my schedule has been successfully parsed.

```
DEBUG: Checking schedule(schedule=before 4am on the first day of the month, tz=null, now=2024-06-27T07:06:12.755Z) (repository=<redacted>, branch=<redacted>)
DEBUG: Checking 1 schedule(s) (repository=<redacted>, branch=<redacted>)
DEBUG: Checking schedule "before 4am on the first day of the month" (repository=<redacted>, branch=<redacted>)
       "parsedSchedule": {"schedules": [{"t_b": [14400], "D": [1]}], "exceptions": [], "error": -1}
DEBUG: Package not scheduled (repository=<redacted>, branch=<redacted>)
DEBUG: Branch + PR exists but is not scheduled -- will update if necessary (repository=<redacted>, branch=<redacted>)
DEBUG: Branch already exists (repository=<redacted>, branch=<redacted>)
```

Am I right in assuming the scheduling functionality is only there to control creating new PRs? If so, is there anyway to also control the frequency of which an existing PR is bumped/updated?

My `renovate.json` is:
```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "kubernetes": {
        "fileMatch": [
            "manifests/.+\\.yaml$"
        ]
    },
    "packageRules": [
        {
            "matchDatasources": [
                "docker"
            ],
            "matchPackagePatterns": [
                "<redacted>"
            ],
            "schedule": [
                "before 4am on the first day of the month"
            ]
        }
    ]
}

```

**Comment by RahulGautamSingh** [comment]
You can use the [`rebaseWhen`](https://docs.renovatebot.com/configuration-options/#rebasewhen) to control the frequency of updates to an existing PR.

Related reading: https://docs.renovatebot.com/updating-rebasing/#updating-and-rebasing-branches

**Reply by RahulGautamSingh** [reply, marked as answer]
And to anwer your first question. Yes, we update the existing branches even when not scheduled. You can control this behaviour using this option [updateNotScheduled](https://docs.renovatebot.com/configuration-options/#updatenotscheduled)
